cpp/src/replace/replace.cu

```diff
@@ -126,41 +126,42 @@ __global__ void replace_strings_first_pass(cudf::column_device_view input,
                                            cudf::bitmask_type* output_valid,
                                            cudf::size_type* __restrict__ output_valid_count)
 {
-  cudf::size_type nrows = input.size();
-  cudf::size_type i     = blockIdx.x * blockDim.x + threadIdx.x;
-  uint32_t active_mask  = 0xffff'ffffu;
-  active_mask           = __ballot_sync(active_mask, i < nrows);
+  cudf::size_type nrows       = input.size();
+  cudf::thread_index_type tid = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
This should be
```suggestion
  cudf::thread_index_type tid = blockIdx.x * blockDim.x + cudf::thread_index_type{threadIdx.x};
```
so we promote before adding the stride?

Reply:
I think promoting `blockIdx.x` first would be the best move. That way the multiplication is done in `int64_t` (which gets the addition, too).
```suggestion
  cudf::thread_index_type tid = static_cast<cudf::thread_index_type>(blockIdx.x) * blockDim.x + threadIdx.x;
```

Reply:
@vuule Do you agree with that analysis?

Reply:
We're wandering close to the shores of the nitpick sea, but... could `blockIdx.x * blockDim.x` ever overflow? I assume we don't launch kernels with blocks that are completely outside of the row range. Also, the type is called `thread_index_type`, to me it looks odd applied to the block index.

Reply:
This is what I'm going with 
https://github.com/rapidsai/cudf/pull/13910/files#diff-c0d67e637bd07fdcf4833b0da7fb7aa7aebd6f992736416a00d86f9c5e6ee32c
would be nice to align somehow :D

Reply:
Promoting just the addend does not seem to be enough to promote the multiplication
https://godbolt.org/z/coqPKqfWe

Reply:
sure, but when do we need to promote the multiplication?

Reply:
Sorry, I thought the point of going with `thread_index_type` was for this very purpose.
Otherwise, we should be ok with `size_type` or perhaps `uint32_t` if the overflow never actually occurs?
Maybe I misunderstood what we are doing this for.

Reply:
Per offline discussion, it seems like I'm just not forward-looking wrt `size_type` size.

Reply:
This is resolved by #13996 and associated utility functions.